docs/aws/audit/lambdamonitoring/rules/has_tags.mdx

```diff
@@ -24,6 +24,295 @@ CBP
+   ```
+
+By following these steps, you can ensure that all your AWS Lambda functions are created with the required tags and that existing functions are checked and updated accordingly.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the AWS Lambda service by typing 'Lambda' into the search bar and selecting 'Lambda' from the dropdown menu.
+3. Once you are in the AWS Lambda console, you will see a list of all your Lambda functions. Click on the name of the function you want to check for tags.
+4. In the function's detail page, scroll down to the 'Tags' section. If there are no tags listed, then the Lambda function does not have any tags.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to list all your Lambda functions. Open your terminal and run the following AWS CLI command:
+
+```bash
+aws lambda list-functions --region your-region
+```
+Replace "your-region" with your actual AWS region. This command will return a list of all your Lambda functions in the specified region.
+
+2. The output will be in JSON format. Each function will have an "FunctionName" field. You can use this field to get more information about each function. 
+
+3. To check if a function has tags, use the following command:
+
+```bash
+aws lambda list-tags --resource your-function-arn
+```
+Replace "your-function-arn" with the actual ARN of your Lambda function. This command will return a list of all tags associated with the specified function.
+
+4. If the output is empty, it means the function doesn't have any tags. If the output contains key-value pairs, these are the tags of your function. Repeat steps 3 and 4 for each function in your list.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3) in your local environment. This will allow you to interact with AWS services including Lambda. You can install it using pip:
+
+```python
+pip install boto3
+```
+
+2. Import the necessary modules and create a session using your AWS credentials:
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='us-west-2'
+)
+```
+
+3. Create a client for AWS Lambda:
+
+```python
+lambda_client = session.client('lambda')
+```
+
+4. Now, you can list all your Lambda functions and check if they have tags. If a function doesn't have tags, print its name:
+
+```python
+def check_lambda_tags():
+    functions = lambda_client.list_functions()
+    for function in functions['Functions']:
+        tags = lambda_client.list_tags(Resource=function['FunctionArn'])
+        if not tags['Tags']:
+            print(f"Lambda function {function['FunctionName']} does not have tags.")
+
+check_lambda_tags()
+```
+
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>